### ISSUE TITLE
REF: allow to disable referencing code, to suppress warnings

### ISSUE THREAD

**Issue by ankostis** [lnoor/sphinx-jsonschema#40]
It is common for JSON documents to define and `$ref` to a `/definitions` section within themselves.
But that habit does not bond very well with this useful extension (it needs a separate directive for each referenced definition, duplicating the work).  Therefore i leave those internal references broken, and use this extension only for linking to external documents, rendering descriptions and presenting the document-structure.

Now in all my projects i have a test-case demanding that no warnings are raised when building the site.
For this extension i have to [suppress too many warnings](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-suppress_warnings), irrelevant to this extension:
```python
 suppress_warnings = ["ref.ref"]
```
Would it be hard to add a configuration-option to disable completely the linking code (even if that breaks external documents)?

**Comment by lnoor** [comment]
Am I correct that by

> i leave those internal references broken

you mean that the `$ref`s in your documents are not matched against `$$target`s?
So your comments in https://github.com/lnoor/sphinx-jsonschema/issues/32#issuecomment-654441471 are expressing some frustration...

I think it wouldn't be to hard to implement a switch.
There are only three lines of code to skip (72-74) in wide_format.py.
And in https://github.com/lnoor/sphinx-jsonschema/commit/e7ab0efa42370a8e2ddf52fa83d02a0d960e301f I implemented (and later removed) the `display_if` switch which also acted as a toggle.

You're sure you don't want to see how it works when splitting up your definitions?

It is a fairly small change you can implement yourself if you want.
I accept a PR for it.
Otherwise I can take care of it but it will be a while before it is on pypi.

**Comment by ankostis** [comment]
sorry for the delayed answer: yes

**Comment by lnoor** [comment]
Hope and believe that the PR by Wouter Tuinstra is going to solve this.
I' ve been testing it and it looks really good so I hope to release it soon.
